pennylane/devices/experimental/execution_config.py

```diff
@@ -41,6 +41,9 @@ class ExecutionConfig:
     shots: Optional[Union[int, Tuple[int]]] = None
     """The number of shots for an execution"""
 
+    grad_on_execution: Union[bool, str] = "best"
+    """Whether or not to compute the gradient at the same time as the execution."""
+
```

Comment:
What will be the interplay between this new control variable and `execute`, `compute_derivatives` and `execute_and_compute_derivatives` methods?

Reply:
If `grad_on_execution` is `True`, then `execute_and_compute_derivatives` will be called. Otherwise, `execute` will first be called, followed by `compute_derivatives` at a later point in time. This is the "gradient mode" referenced by #2188 .